scripts/download_pypi_packages.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+import os
+import json
+import shutil
+
+from typing import Dict, Any
+from urllib.request import urlretrieve
+
+argparser = argparse.ArgumentParser(
+    prog="download_pypi_packages", description="Helper program to download PyPI packages",
+)
+argparser.add_argument(
+    "-n", "--number", type=int, default=100, help="Number of packages to download"
+)
+argparser.add_argument(
+    "-a", "--all", action="store_true", help="Download all packages listed in the json file"
+)
+
+
+def load_json(filename: str) -> Dict[Any, Any]:
+    with open(os.path.join("data", f"{filename}.json"), "r") as f:
+        j = json.loads(f.read())
+    return j
+
+
+def remove_json(filename: str) -> None:
+    path = os.path.join("data", f"{filename}.json")
+    os.remove(path)
+
+
+def download_package_json(package_name: str) -> None:
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    urlretrieve(url, os.path.join("data", f"{package_name}.json"))
+
+
+def download_package_code(name: str, package_json: Dict[Any, Any]) -> None:
+    source_index = -1
+    for idx, url_info in enumerate(package_json["urls"]):
+        if url_info["python_version"] == "source":
+            source_index = idx
+            break
+    filename = package_json["urls"][source_index]["filename"]
+    url = package_json["urls"][source_index]["url"]
+    urlretrieve(url, os.path.join("data", filename))
```

Comment:
Can you use a different directory than `data`? I want to be able to clean out the entire directory with a simple command, but there are also git-controlled files under `data`.